### DISCUSSION TITLE
[Wasm] ListView GroupStyle.HeaderTemplate not rendering

### DISCUSSION THREAD

**Discussion by bricelam** [unoplatform/uno#8614] [Q&A] [answered]
### Current behavior

Group headers are not shown on Web Assembly.

![image](https://user-images.githubusercontent.com/1226749/164945872-564c8729-c0f7-4c51-8866-4da40cf72da2.png)

### Expected behavior

They should render the same as the do on Android...

![image](https://user-images.githubusercontent.com/1226749/164945753-64051a64-0ac0-4bfd-919d-75f9da9f9208.png)

...or UWP.

![image](https://user-images.githubusercontent.com/1226749/164945818-072fc2f1-e1f3-46e7-a22c-26f503447832.png)

### How to reproduce it (as minimally and precisely as possible)

I'm using this in my view...

```xaml
<Page.Resources>
  <CollectionViewSource x:Name="FruitsCVS" Source="{x:Bind Fruits}" IsSourceGrouped="True" />
</Page.Resources>

<ListView ItemsSource="{x:Bind FruitsCVS.View}">
  <ListView.GroupStyle>
    <GroupStyle>
      <GroupStyle.HeaderTemplate>
        <DataTemplate>
          <TextBlock Text="{Binding Key}" />
        </DataTemplate>
      </GroupStyle.HeaderTemplate>
    </GroupStyle>
  </ListView.GroupStyle>
</ListView>
```

...and this in my code-behind.

```cs
public IEnumerable<IGrouping<string, string>> Fruits { get; }
    = new[] { "Apple", "Apricot", "Banana" }
        .GroupBy(f => f.Substring(0, 1))
        .OrderBy(g => g.Key);
```

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report! This is actually an issue tracked here: https://github.com/unoplatform/uno/issues/234. @dr1rrb may have some more information to provide here as a workaround using `ItemsRepeater`.

**Reply by dr1rrb** [reply]
Well a solution would indeed be to use the `ItemsRepeater`. For grouping, you just have to nest `ItemsRepeater`s eg:
```xaml

<Page.Resources>
  <DataTemplate x:Key="FruitTemplate">
    <TextBlock Text="{Binding}" />
  </DataTemplate>
</Page.Resources>

<ScrollViewer>
  <ItemsRepeater ItemsSource="{x:Bind FruitsCVS.View}">
    <ItemsRepeater.ItemTemplate>
      <DataTemplate>
        <StackPanel>
          <TextBlock Text="{Binding Key}" />
          <ItemsRepeater ItemsSource="{Binding}" ItemTemplate="{StaticResource FruitTemplate}">
        </StackPanel>
      </DataTemplate>
    </ItemsRepeater.ItemTemplate>
  </ItemsRepeater>
</ScrollViewer>
```

**Reply by dr1rrb** [reply]
Here in live: https://playground.platform.uno/#92bb9f16

**Comment by KWodarczyk** [comment]
Same issue with GridView, which is more complex control than ListView so not sure using nested repeaters will do the job,  any idea when this be fixed ? It works fine on Windows Apps Sdk target but not in Web Assembly

**Reply by MartinZikmund** [reply]
Is the use of `ItemsRepeater` a valid solution for you?

**Reply by KWodarczyk** [reply]
It's ok for time being
